src/main/java/org/scijava/convert/StringToArrayConverter.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * #%L
+ * SciJava Common shared library for SciJava software.
+ * %%
+ * Copyright (C) 2009 - 2022 SciJava developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.convert;
+
+import org.scijava.Priority;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.util.Types;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A {@link Converter} that specializes in converting {@link String}s to
+ * n-dimensional arrays. This {@link Converter} can convert any array whose
+ * component types can be created from a {@link String}. By default, this
+ * {@link Converter} delimits the {@link String} based on commas.
+ * 
+ * @author Gabriel Selzer
+ */
+@Plugin(type = Converter.class, priority=Priority.VERY_LOW)
+public class StringToArrayConverter extends AbstractConverter<String, Object> {
+
+	@Parameter
+	private ConvertService convertService;
+
+	@Override
+	public boolean canConvert(final Class<?> src, final Class<?> dest) {
+		if (src == null) return false;
+		final Class<?> saneSrc = Types.box(src);
+		final Class<?> saneDest = Types.box(dest);
+		return saneSrc == String.class && saneDest.isArray();
+	}
+
+	@Override
+	public boolean canConvert(final Object src, final Class<?> dest) {
+		if (!canConvert(src.getClass(), dest)) return false;
+		String srcString = (String) src;
+		if (!(srcString.startsWith("{") && srcString.endsWith("}"))) return false;
+		List<String> components = elements((String) src);
+		return components.size() == 0 || convertService.supports(components.get(0),
```

Comment:
Good point, thanks.

> But I think it's an acceptable compromise, since the only way to be completely sure would be to actually do the full conversion here.

We *could* consider doing the conversion here, though, and caching it for later in a `WeakHashMap`...but I feel like that could have its own issues, e.g. with repeated calls to `convert` returning the same object...so I'm not going to implement this.